src/main/java/bio/terra/service/snapshot/flight/create/SnapshotCreateFlight.java

```diff
@@ -292,6 +295,16 @@ public SnapshotCreateFlight(FlightMap inputParameters, Object applicationContext
     // The IAM code contains retries, so we don't make a retry rule here.
     addStep(new SnapshotAuthzIamStep(iamClient, snapshotService, snapshotReq, userReq, snapshotId));
 
+    // add data access control groups to the snapshot
+    List<String> dataAccessControlGroups = snapshotReq.getDataAccessControlGroups();
+    if (Objects.nonNull(dataAccessControlGroups) && !dataAccessControlGroups.isEmpty()) {
+      addStep(
+          new CreateSnapshotGroupConstraintPolicyStep(
+              policyService, snapshotId, dataAccessControlGroups));
+      addStep(
+          new AddSnapshotAuthDomainStep(iamService, userReq, snapshotId, dataAccessControlGroups));
```

Comment:
do we need to add the journal step as well? Or no, because at the end of the snapshot create flight when the snapshot is journaled it will include the dac?